Mathlib/Data/FP/Basic.lean

```diff
@@ -0,0 +1,293 @@
+/-
+Copyright (c) 2017 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+
+! This file was ported from Lean 3 source module data.fp.basic
+! leanprover-community/mathlib commit 7b78d1776212a91ecc94cf601f83bdcc46b04213
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Semiquot
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Implementation of floating-point numbers (experimental).
+-/
+
+-- Porting note: TODO add docs and remove `@[nolint docBlame]`
+
+@[nolint docBlame]
+def Int.shift2 (a b : ℕ) : ℤ → ℕ × ℕ
+  | Int.ofNat e => (a.shiftl e, b)
+  | Int.negSucc e => (a, b.shiftl e.succ)
+#align int.shift2 Int.shift2
+
+namespace FP
+
+@[nolint docBlame]
+inductive RMode
+  | NE -- round to nearest even
+  deriving Inhabited
+#align fp.rmode FP.RMode
+
+@[nolint docBlame]
+class FloatCfg where
+  (prec emax : ℕ)
+  PrecPos : 0 < prec
+  PrecMax : prec ≤ emax
```

Comment:
These should in fact start with lowercase. Only types (and props) start with a capital letter.
```suggestion
  precPos : 0 < prec
  precMax : prec ≤ emax
```